### ISSUE TITLE
[boring-sys] Release v2.1.0 bindgen failure with clang16

### ISSUE THREAD

**Issue by BiagioFesta** [cloudflare/boring#129]
Released `boring-sys` crate (`v2.1.0`) does *not* compile with *llvm-16* (tested on *centos8*).

This seems due to [already fixed bug](https://github.com/rust-lang/rust-bindgen/issues/2312) in `bindgen`.

---

Error on `boring-sys`:
```
error: failed to run custom build command for `boring-sys v2.1.0 (/root/boring/boring-sys)`
Caused by:
  process didn't exit successfully: `/root/boring/target/debug/build/boring-sys-dff89fb02913e519/build-script-build` (exit status: 101)
[...]

 --- stderr
  thread 'main' panicked at '"__pthread_cond_s_union_(anonymous_at_/usr/include/bits/thread-shared-types_h_173_17)" is not a valid Ident', /root/.cargo/registry/src/index.crates.io-6f17d22bba15001f/proc-macro2-1.0.66/src/fallback.rs:774:9
[...]
```

**Notes**:
 * The issue is reproducible only with `v2.1.0` released version. By using `master` it successfully compiles.
   - That's considering `master` uses a fixed version of `bindgen`.
 * The issue seems to be only reproducible with llvm16 (by downgrading to version 15, even `v2.1.0` compiles).

### To reproduce

<details>
  <summary>Dockerfile</summary>
  
  ```Dockerfile
FROM quay.io/centos/centos:stream8

ENV BORING_BRANCH="v2.1.0"
ENV CLANG_VERSION="16.0.0"

RUN dnf -y update
RUN dnf -y install epel-release
RUN dnf -y update
RUN dnf -y install git gcc gcc-c++ cmake
RUN dnf -y install clang-devel-${CLANG_VERSION}
RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y

WORKDIR /root

RUN git clone --depth 1 --branch ${BORING_BRANCH} https://github.com/cloudflare/boring.git
  ```
</details>

---

Considering the current public release available on crates.io (the main artifact source for this) is still `v2.1.0` for the `boring` crate, I think it was still worth it to open this issue to track this.

Moreover, is there any plan for a new release to unblock this?

Thanks.

**Comment by inikulin** [comment]
duplicate of #124
